### PR TITLE
[UNR-4414] Gameplay cues gym

### PR DESCRIPTION
**Pull request by KeyMaster-**
Adds a character with a properly set up ability system component, and abilities/effects/cues etc to test gameplay cue activation.

Blueprint logic:
GameplayCuesCharacter_BP: Adds an effect that grants Ability_ApplyAddGC and Ability_ApplyExecuteGC to the character's ASC. Also activates the execute/added GC abilities on Q/E respectively
image

Ability_ApplyExecuteGC/Ability_ApplyAddGC: Both do the same logic of adding a gameplay effect to the character, but with different effects, Effect_ExecuteGC and Effect_AddGC respectively. The abilities are configured to be client-predicted.
2020-11-04 14_09_48-GDKTestGyms - Unreal Editor

Cue_Execute: Spawns an explosion particle system at the target actor's location.
2020-11-04 14_10_10-GDKTestGyms - Unreal Editor

### CODE REVIEW

File: Game/Source/GDKTestGyms/Public/GasCharacter.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+/**
+ * A GDK Test Gyms Character with an Ability System Component added and set up to correctly initialise.
+ */
+UCLASS()
+class GDKTESTGYMS_API AGasCharacter : public AGDKTestGymsCharacter, public IAbilitySystemInterface
+{
+	GENERATED_BODY()
+	
+	//Implement IAbilitySystemInterface.
```
Comment (m-samiec) [inline Game/Source/GDKTestGyms/Public/GasCharacter.h:19]:
```suggestion
	// Implement IAbilitySystemInterface.
```

File: Game/Source/GDKTestGyms/Public/GasCharacter.h [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	UAbilitySystemComponent* GetAbilitySystemComponent() const override	
+	{
+		return AbilitySystemComponent;
+	}
+
+public:
+	AGasCharacter();
+
+	UPROPERTY(VisibleAnywhere, BlueprintReadOnly, Category = Abilities)
+	class UAbilitySystemComponent* AbilitySystemComponent;
```
Comment (m-samiec) [inline Game/Source/GDKTestGyms/Public/GasCharacter.h:29]:
```suggestion
	UAbilitySystemComponent* AbilitySystemComponent;
```

File: USER_MANUAL.md [hunk 3]
```diff
@@ -289,5 +289,14 @@ The ReplicatedStartupActorTest is failing, pending https://improbableio.atlassia
   * MaxEventTracingFileSizeBytes=104857600
 * You will probably want to remove these setting after you have finished running these tests.
 
+##### Gameplay Cues gym
+* Tests that gameplay cues get correctly activated on all clients.
+* Includes both a non-instanced gameplay cue that gets executed, and an instanced cue that gets added and removed as a gameplay effect with duration is added and removed.
+* How to test:
+  * Run the map with multiple clients.
+  * Position one client's character in view of the other client.
+  * Press Q to trigger the executed gameplay cue. A burst of sparks should be emitted from the controlled character, which should also be visible on the other client.
```
Comment (m-samiec) [inline USER_MANUAL.md:298]: Could we add some print/logs so it's more explicit if the test is successful?

**Comment by Vix2021** [comment]
Looks good but I am not familiar with GAS.
It could be useful to add an automated functional test for gameplay cue activation but perhaps that is outside the scope of this ticket.

**Review by m-samiec — APPROVED** [review]

**Pull request merged**